### PR TITLE
feat: add missing with_counts parameter to fetch_guilds method

### PR DESCRIPTION
**Pull request by BruhDark**
## Summary

This PR adds `with_counts` for the `fetch_guilds` method, which was missing.

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Comment by BruhDark** [comment]
Side note 1: Blame my IDE for "fixing" spaces in other funcs in the files I edited.
Side note 2: Might wanna double check the edited documentation on such methods, don't know if it is correct.

**Review by plun1331 — COMMENTED** [review]
Changelog please :3 (under Added)

File: discord/client.py [hunk 1]
```diff
@@ -1489,6 +1490,11 @@ def fetch_guilds(
             Retrieve guilds after this date or object.
             If a datetime is provided, it is recommended to use a UTC aware datetime.
             If the datetime is naive, it is assumed to be local time.
+        with_counts: Optional[:class:`bool`]
```
Comment (Dorukyum) [inline discord/client.py:1493]: Shouldn't be None.
```suggestion
        with_counts: :class:`bool`
```

File: discord/iterators.py [hunk 2]
```diff
@@ -585,9 +585,14 @@ class GuildIterator(_AsyncIterator["Guild"]):
         Object before which all guilds must be.
     after: Optional[Union[:class:`abc.Snowflake`, :class:`datetime.datetime`]]
         Object after which all guilds must be.
+    with_counts: Optional[:class:`bool`]
```
Comment (Dorukyum) [inline discord/iterators.py:588]:
```suggestion
    with_counts: :class:`bool`
```

**Review by Dorukyum — CHANGES_REQUESTED** [review]

**Comment by BruhDark** [comment]
Done

**Review by plun1331 — DISMISSED** [review]

File: discord/client.py [hunk 3]
```diff
@@ -1489,6 +1490,11 @@ def fetch_guilds(
             Retrieve guilds after this date or object.
             If a datetime is provided, it is recommended to use a UTC aware datetime.
             If the datetime is naive, it is assumed to be local time.
+        with_counts: :class:`bool`
+            Whether to include count information in the guilds. This fills the
```
Comment (JustaSqu1d) [inline discord/client.py:1494]:
```suggestion
            Whether to include member count information in guilds. This fills the
```

File: discord/iterators.py [hunk 4]
```diff
@@ -585,9 +585,14 @@ class GuildIterator(_AsyncIterator["Guild"]):
         Object before which all guilds must be.
     after: Optional[Union[:class:`abc.Snowflake`, :class:`datetime.datetime`]]
         Object after which all guilds must be.
+    with_counts: :class:`bool`
+        Whether to include count information in the guilds. This fills the
```
Comment (JustaSqu1d) [inline discord/iterators.py:589]:
```suggestion
        Whether to include member count information in guilds. This fills the
```

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]

**Comment by BruhDark** [comment]
Squid and his grammar.

**Review by plun1331 — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]

**Review by plun1331 — DISMISSED** [review]

**Review by plun1331 — APPROVED** [review]

**Pull request merged**